### PR TITLE
API: BufList improvements + new TabList implementation

### LINKED ISSUE #163: BufferList and TabList API
Hi,
it would be great if the API can provide a BufferList with id, name, visible and current attributes for each buffer. Also having a TabList method with id, name and current attribute for each tab would be great.

### PR DESCRIPTION
**Pull request by sappo**
See commit messages for details.

fixes #163

### CODE REVIEW

**Comment by sappo** [comment]
@szw ping

Any comments are appreciated ;) Thanks!

**Comment by sappo** [comment]
@szw ping again.

**Comment by sappo** [comment]
@szw I really like vim-ctrlspace but no getting my PR merged and even worse not getting a response at all is really annoying. So if you don't want contributers your on a road to success :-(

File: autoload/ctrlspace/api.vim [hunk 1]
```diff
@@ -234,6 +236,26 @@ function! ctrlspace#api#Guitablabel()
...
+	let currentTab = tabpagenr()
+
+	for t in range(1, lastTab)
+		let winnr       = tabpagewinnr(t)
+		let buflist     = tabpagebuflist(t)
+		let bufnr       = buflist[winnr - 1]
+		let bufname     = bufname(bufnr)
+		let tabTitle    = ctrlspace#api#TabTitle(t, bufnr, bufname)
+		let tabModified = ctrlspace#api#TabModified(t)
+                let tabCurrent  = t == currentTab
```
Comment (szw) [inline autoload/ctrlspace/api.vim]: there is some issue with indentation here

**Comment by szw** [comment]
I'm sorry for the delay :(. I was seriously busy with other projects. Your pull request looks really well :+1:. The `BufferList` function is used internally only for moving the bar. And it this particular case, later we could develop some more efficient internal one.

File: autoload/ctrlspace/api.vim (hunk 1 above)
Comment (sappo) [inline autoload/ctrlspace/api.vim]: I'll fix some time later today.

**Comment by sappo** [comment]
Great, thanks for merging :-)

**Pull request merged**